presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PreferredProperties.java

```diff
@@ -96,6 +100,70 @@ public static PreferredProperties local(List<? extends LocalProperty<Symbol>> lo
         return localProperties;
     }
 
+    public static PreferredProperties derivePreferences(PreferredProperties parentProperties,
+            Set<Symbol> partitioningColumns,
+            List<LocalProperty<Symbol>> localProperties)
+    {
+        // Build identity translations
+        ImmutableMap.Builder<Symbol, Symbol> translations = ImmutableMap.builder();
+        partitioningColumns.forEach(sym -> translations.put(sym, sym));
```

Comment:
Also, you can do this with Java 8 streams like this:

```java
Map<Symbol, Symbol> translations = partitioningColumns.stream().collect(Collectors.toMap(item -> item, item -> item));
```

As a general rule, try to avoid using side-effects for things that can be modeled as pure functions.